pkg/apis/autoscaling/annotation_validation_test.go

```diff
@@ -257,6 +257,26 @@ func TestValidateAnnotations(t *testing.T) {
 		name:        "invalid format and valid class for /window annotation",
 		annotations: map[string]string{WindowAnnotationKey: "jerry-was-a-racecar-driver", ClassAnnotationKey: KPA},
 		expectErr:   "invalid value: jerry-was-a-racecar-driver: " + WindowAnnotationKey,
+	}, {
+		name:        "progressDeadline invalid",
+		annotations: map[string]string{ProgressDeadlineAnnotationKey: "jerry-was-a-racecar-driver"},
+		expectErr:   "invalid value: jerry-was-a-racecar-driver: " + ProgressDeadlineAnnotationKey,
+	}, {
+		name:        "progressDeadline too precise",
+		annotations: map[string]string{ProgressDeadlineAnnotationKey: "1m9s82ms"},
+		expectErr:   "must be specified with at most second precision: " + ProgressDeadlineAnnotationKey,
+	}, {
+		name:        "annotation /progressDeadline is valid for class KPA",
+		annotations: map[string]string{ProgressDeadlineAnnotationKey: "7s", ClassAnnotationKey: KPA},
```

Comment:
nit: is the `ClassAnnotationKey` necessary?

Reply:
I don't think so... it was one of the `/window` tests, so I just brought it along. Will drop it.